subsys/bluetooth/host/conn.c

```diff
@@ -1770,11 +1779,18 @@ int bt_conn_disconnect(struct bt_conn *conn, u8_t reason)
 
 	switch (conn->state) {
 	case BT_CONN_CONNECT_SCAN:
+	case BT_CONN_CONNECT_DIR_ADV:
 		conn->err = reason;
 		bt_conn_set_state(conn, BT_CONN_DISCONNECTED);
 		if (IS_ENABLED(CONFIG_BT_CENTRAL)) {
 			bt_le_scan_update(false);
 		}
+		if (IS_ENABLED(CONFIG_BT_PERIPHERAL)) {
+			/* User should unref connection object when receiving
+			 * error in connection callback.
+			 */
+			return bt_le_adv_stop();
```

Comment:
Are you stopping advertising on disconnect of master role?

Reply:
This looks suspicious to me as well. Shouldn't the bt_le_adv_stop() only be done for BT_CONN_CONNECT_DIR_ADV? Now it's done also for BT_CONN_CONNECT_SCAN, which can't be right, ban it? Remember that the app may have both CONFIG_BT_PERIPHERAL=y and CONFIG_BT_CENTRAL=y.